source/slang/slang-lower-to-ir.cpp

```diff
@@ -6006,7 +6010,7 @@ struct StmtLoweringVisitor : StmtVisitor<StmtLoweringVisitor>
         // a (non-`void`) value, and one for returning "no value"
         // (which effectively returns a value of type `void`).
         //
-        if( auto expr = stmt->expression )
+        if (auto expr = stmt->expression)
```

Comment:
We also need to handle the case where `expr==nullptr`, and we are inside a ConstructorDecl. In this case, we need to:
1. just `return` if for `NonCopyableType`.
2. emit `return *this;` for ordinary case.

Make sure to comment well.

Reply:
does this solve 2.
`getBuilder()->emitReturn(getSimpleVal(context, context->thisVal));`

Reply:
yes. but we can't do this for 1).

Reply:
right. for 1, I'm simply doing `getBuilder()->emitReturn();`